sdk/core/azure_core/src/credentials.rs

```diff
@@ -29,6 +30,46 @@ impl Secret {
     }
 }
 
+/// Represents a Cosmos DB connection string.
+///
+/// The [`Debug`] implementation will not print the secret.
+#[derive(Clone, Deserialize, Serialize)]
+pub struct CosmosConnectionString {
+    pub account_endpoint: Cow<'static, str>,
+    pub account_key: Cow<'static, str>,
+}
+
+impl CosmosConnectionString {
+    pub fn from_secret(secret: &Secret) -> crate::Result<Self> {
+        let split = secret.secret().split(";");
+        let collection = split.collect::<Vec<&str>>();
+
+        match collection.len() {
+            2 => {
+                if (collection[0].contains("AccountEndpoint=")
+                    && collection[1].contains("AccountKey="))
+                {
+                    Ok(Self {
+                        account_endpoint: collection[0][16..collection[0].len() - 1]
+                            .to_owned()
+                            .into(),
+                        account_key: collection[0][11..collection[0].len() - 1].to_owned().into(),
+                    })
```

Comment:
This parsing assumes too much. You should split key=value pairs and find the ones you need.